app/jobs/demo_mode/account_generation_job.rb

```diff
@@ -5,9 +5,16 @@ class AccountGenerationJob < DemoMode.base_job_name.constantize
     def perform(session, **options)
       session.with_lock do
         persona = session.persona
-        raise "Unknown persona: #{session.persona_name}" if persona.blank?
+        if persona.blank?
+          session.update!(error: "Unknown persona: #{session.persona_name}")
+          raise "Unknown persona: #{session.persona_name}"
+        end
 
-        signinable = persona.generate!(variant: session.variant, password: session.signinable_password, options: options)
+        begin
+          signinable = persona.generate!(variant: session.variant, password: session.signinable_password, options: options)
+        rescue StandardError => e
+          session.update!(error: e.message)
+        end
         session.update!(signinable: signinable)
```

Comment:
If an exception is raised by `persona.generate!`, this line will still be executed. Is that intended?